onnxruntime/core/framework/device_stream_collection.cc

```diff
@@ -9,15 +9,9 @@ namespace onnxruntime {
 
 class DeviceStreamCollectionImpl {
  public:
-  DeviceStreamCollectionImpl(size_t num_streams, const SessionState& sess_state) : num_streams_(num_streams) {
+  DeviceStreamCollectionImpl(size_t num_streams, const std::map<OrtDevice, AllocatorPtr>& allocators, bool is_main_graph) : num_streams_(num_streams), allocators_(allocators), is_main_graph_(is_main_graph) {
```

Comment:
>const std::map<OrtDevice, AllocatorPtr>& allocat [](http://example.com/codeflow?start=49&length=48)

let's sync offline on this. it might become more complicated for training, as we are sharing the device stream collection across multiple session